webapp/channels/src/components/admin_console/access_control/editors/table_editor/table_editor.tsx

```diff
@@ -89,204 +75,279 @@ const parseExpression = async (expr: string): Promise<TableRow[]> => {
+                if (newValues.length > 1) {
+                    newValues = [newValues[0]];
+                }
+            }
 
-    const updateRowValues = (index: number, values: string[]) => {
-        const newRows = [...rows];
-        newRows[index].values = values;
-        setRows(newRows);
-        updateExpression(newRows);
-    };
+            const newRows = [...currentRows];
+            newRows[index] = {
+                ...currentRows[index],
+                operator: newOperator,
+                values: newValues,
+            };
 
-    // Get available attributes (excluding ones already used)
-    const getAvailableAttributes = () => {
-        const usedAttributes = new Set(rows.map((row) => row.attribute));
-        return userAttributes.filter((attr) => !usedAttributes.has(attr.attribute));
-    };
+            updateExpression(newRows);
+            return newRows;
+        });
+    }, [updateExpression]);
+
+    const updateRowValues = useCallback((index: number, values: string[]) => {
+        setRows((currentRows) => {
+            const newRows = [...currentRows];
+            newRows[index] = {...newRows[index], values};
+            updateExpression(newRows);
+            return newRows;
+        });
+    }, [updateExpression]);
 
     return (
         <div className='table-editor'>
-            <div className='table-editor__table'>
-                <div className='table-editor__header'>
-                    <div className='table-editor__column-header'>
-                        <FormattedMessage
-                            id='admin.access_control.table_editor.attribute'
-                            defaultMessage='Attribute'
-                        />
-                    </div>
-                    <div className='table-editor__column-header'>
-                        <FormattedMessage
-                            id='admin.access_control.table_editor.operator'
-                            defaultMessage='Operator'
-                        />
-                    </div>
-                    <div className='table-editor__column-header'>
-                        <FormattedMessage
-                            id='admin.access_control.table_editor.values'
-                            defaultMessage='Values'
-                        />
-                    </div>
-                    <div className='table-editor__column-header-actions'/>
-                </div>
-
-                <div className='table-editor__rows'>
-                    {rows.length === 0 ? (
-                        <div className='table-editor__blank-state'>
-                            <span>
-                                {formatMessage({
-                                    id: 'admin.access_control.table_editor.blank_state',
-                                    defaultMessage: 'Select a user attribute and values to create a rule',
-                                })}
+            <table className='table-editor__table'>
```

Comment:
> Similar to my comment on the job table, if we rewrite this using a table element, it'll make positioning everything much easier. This is currently using nested flexboxes to recreate a table in a way that doesn't seem necessary

https://github.com/mattermost/mattermost/pull/30785#discussion_r2089062240 comment from @hmhealey 

Started to use table, it really helped with the alignment issues, thanks!